asset/src/kafka_sender/processor.ts

```diff
@@ -6,45 +6,110 @@ import {
     ConnectionConfig,
     getValidDate,
     isString,
+    has,
 } from '@terascope/job-components';
 import * as kafka from 'node-rdkafka';
 import { KafkaSenderConfig } from './interfaces';
 import { ProducerClient, ProduceMessage } from '../_kafka_clients';
 
-export default class KafkaSender extends BatchProcessor<KafkaSenderConfig> {
+interface Endpoint {
     producer: ProducerClient;
+    data: any[];
+}
+
+interface TopicMap {
```

Comment:
If we are calling this a Map, why don't we just make it a `Map`?

```suggestion
type TopicMap = Map<string, Endpoint>;
```